MiniDumper/Debugger.cs

```diff
@@ -48,41 +53,72 @@ void TakeDumps()
             ShowBanner();
 
             // setup Ctrl+C listener
-            Console.CancelKeyPress += (o, ev) => {
+            Console.CancelKeyPress += (o, ev) =>
+            {
                 Console.WriteLine("Ctrl + C received - detaching from a process");
                 ev.Cancel = true;
-                _shouldDeatch = true; 
+                _shouldDeatch = true;
             };
+
             WaitForDebugEvents();
         }
 
         private void WaitForDebugEvents()
         {
-            using (var miniDumper = CreateMiniDumper()) {
-                if (_options.NoDumpOptionSelected) {
+            using (var miniDumper = CreateMiniDumper())
+            {
+                if (_options.NoDumpOptionSelected)
+                {
                     miniDumper.DumpWithoutReason();
                     DetachProcess();
                     return;
                 }
 
-                while (!_detached) {
+                if (_options.MemoryCommitThreshold.HasValue || _options.MemoryCommitDrops.HasValue)
+                {
+                    bool isTimerSet = false;
+                    if (!_options.NeedAttachDebugger)
+                    {
+                        DetachProcess();
+
+                        while (!IsPollingCompleted)
+                        {
+                            if (!isTimerSet)
```

Comment:
Even for single-line conditions please use parenthesis - they are easier to spot and that's the convention we use in minidumper code. There are few pleaces in the code where it needs to be fixed